src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/Symbolic/SymbolicRegexMatcher.cs

```diff
@@ -183,13 +206,37 @@ private SymbolicRegexMatcher(SymbolicRegexBuilder<TSet> builder, SymbolicRegexNo
                 _positionKinds[mintermId + 1] = CalculateMintermIdKind(mintermId);
             }
 
+            // Create optimized reversal
+            _optimizedReversalState = CreateOptimizedReversal(_pattern.Reverse(builder));
+
             // Store the find optimizations that can be used to jump ahead to the next possible starting location.
             // If there's a leading beginning anchor, the find optimizations are unnecessary on top of the DFA's
             // handling for beginning anchors.
             if (findOptimizations.IsUseful &&
                 findOptimizations.LeadingAnchor is not RegexNodeKind.Beginning)
             {
-                _findOpts = findOptimizations;
+                // In some cases where the findOptimizations are useful, just using the DFA can still be faster.
+                _findOpts = findOptimizations switch
+                {
+                    // string literals are the best case
+                    {
+                        FindMode:
+                        FindNextStartingPositionMode.LeadingString_LeftToRight or
+                        FindNextStartingPositionMode.FixedDistanceString_LeftToRight or
+                        FindNextStartingPositionMode.LeadingString_OrdinalIgnoreCase_LeftToRight
+                    } => findOptimizations,
+                    // note: only the Teddy implementation is faster than DFA here, Aho Corasick should map to null
+                    {
+                        FindMode:
+                        FindNextStartingPositionMode.LeadingStrings_LeftToRight or
+                        FindNextStartingPositionMode.LeadingStrings_OrdinalIgnoreCase_LeftToRight,
+                        LeadingStrings: not null
+                    } when findOptimizations.LeadingStrings.GetType().Name != "StringSearchValuesAhoCorasick`2" => findOptimizations,
```

Comment:
One case i wanted to avoid is when there's a significant amount of alternatives. What about disabling based on regex AST size since we compute it at the beginning?

Reply:
> What about disabling based on regex AST size since we compute it at the beginning?

That also seems like 6 of one / half dozen of another. There could be lots of alternations in the middle but with a very simple and searchable beginning?

Reply:
That's true as well. Since it's hard to decide without looking into the implementation itself we can just leave findOptimizations on. I need to do some comparisons on large patterns to see how much it changes